catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/content/data/impl/ContentItemImpl.java

```diff
@@ -108,6 +144,15 @@ public ContentItemImpl(String id, ByteSource byteSource, String mimeTypeRawData,
         } catch (MimeTypeParseException e) {
             LOGGER.debug("Unable to create MimeType from raw data " + mimeTypeRawData);
         }
+        try {
+            if (StringUtils.isNotBlank(this.qualifier)) {
+                uri = new URI(CONTENT_SCHEME, this.id, this.qualifier);
+            } else {
+                uri = new URI(CONTENT_SCHEME + ":" + this.id);
```

Comment:
Instead of concatenation, you can just pass in a null for the fragment and get the correct URI.
```
uri = new URI(CONTENT_SCHEME, this.id, null)
```